### PR TITLE
add additional attributes for OTel compatibility

### LINKED ISSUE #2190: Add new OTel compatible DB / HTTP attributes
Details TBD.

Add support for OTel standard DB attributes to increase compatibility with OTel.

### PR DESCRIPTION
**Pull request by fallwith**
3 attributes have been added to datastore segments and 3 attributes have been added to external request segments for improved alignment with OTel specifications.

resolves #2190

### CODE REVIEW

File: lib/new_relic/agent/span_event_primitive.rb [hunk 1]
```diff
@@ -29,12 +29,16 @@ module SpanEventPrimitive
       CATEGORY_KEY = 'category'
       HTTP_URL_KEY = 'http.url'
       HTTP_METHOD_KEY = 'http.method'
+      HTTP_REQUEST_METHOD_KEY = 'http.request.method'
```
Comment (hannahramadan) [inline lib/new_relic/agent/span_event_primitive.rb:32]: Curious, what could the difference be between `HTTP_METHOD_KEY` and `HTTP_REQUEST_METHOD_KEY`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ Version <dev> adds instrumentation for Async::HTTP, Ethon, and HTTPX, gleans Doc
 
   For compatibility with Ruby 3.4 and to silence compatibility warnings present in Ruby 3.3, declare a dependency on the `base64` gem. The New Relic Ruby agent uses the native Ruby `base64` gem for Base 64 encoding/decoding. The agent is joined by Ruby on Rails ([rails/rails@3e52adf](https://github.com/rails/rails/commit/3e52adf28e90af490f7e3bdc4bcc85618a4e0867)) and others in making this change in preparation for Ruby 3.3/3.4. [PR#2238](https://github.com/newrelic/newrelic-ruby-agent/pull/2238)
 
--**Feature: Add Roda support for the newrelic_ignore\* family of methods**
+- **Feature: Add Roda support for the newrelic_ignore\* family of methods**
```
Comment (hannahramadan) [inline CHANGELOG.md:35]: thank you!

File: lib/new_relic/agent/span_event_primitive.rb (hunk 1 above)
Comment (fallwith) [reply lib/new_relic/agent/span_event_primitive.rb:32]: The attribute names differ but their values are identical. There is a desire to rename `http.method` to `https.request.method`. We're not ready to introduce any backwards compatibility issues by renaming it at this time, so instead we are simply adding the new name. It's possible that we'll now proceed to deprecate the old name and stop using it at a future date.

Comment (hannahramadan) [reply lib/new_relic/agent/span_event_primitive.rb:32]: Makes sense ✨

**Review by hannahramadan — DISMISSED** [review]

**Review by tannalynn — DISMISSED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## dev
 
-Version <dev> adds instrumentation for Async::HTTP, Ethon, and HTTPX, gleans Docker container IDs from cgroups v2-based containers, records additional synthetics attributes, fixes an issue with Rails 7.1 that could cause duplicate log records to be sent to New Relic, and fixes a deprecation warning for the Sidekiq error handler.
+Version <dev> adds instrumentation for Async::HTTP, Ethon, and HTTPX, adds the ability to ignore specific routes with Roda, gleans Docker container IDs from cgroups v2-based containers, records additional synthetics attributes, fixes an issue with Rails 7.1 that could cause duplicate log records to be sent to New Relic, fixes a deprecation warning for the Sidekiq error handler, and adds additional attributes for Open Telemetry compatibility.
```
Comment (kaylareopelle) [inline CHANGELOG.md:5]:
```suggestion
Version <dev> adds instrumentation for Async::HTTP, Ethon, and HTTPX, adds the ability to ignore specific routes with Roda, gleans Docker container IDs from cgroups v2-based containers, records additional synthetics attributes, fixes an issue with Rails 7.1 that could cause duplicate log records to be sent to New Relic, fixes a deprecation warning for the Sidekiq error handler, and adds additional attributes for OpenTelemetry compatibility.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -41,6 +41,22 @@ Version <dev> adds instrumentation for Async::HTTP, Ethon, and HTTPX, gleans Doc
 
   For more information, see [Roda Instrumentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/instrumented-gems/roda-instrumentation/). [PR#2267](https://github.com/newrelic/newrelic-ruby-agent/pull/2267)
 
+- **Feature: Add additional span attributes for Open Telemetry compatibility**
```
Comment (kaylareopelle) [inline CHANGELOG.md:44]:
```suggestion
- **Feature: Add additional span attributes for OpenTelemetry compatibility**
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -41,6 +41,22 @@ Version <dev> adds instrumentation for Async::HTTP, Ethon, and HTTPX, gleans Doc
...
+
+  For improved compatibility with Open Telemetry's specifications, the agent's datastore (for databases) and external request (for HTTP clients) segments have been updated with additional attributes.
```
Comment (kaylareopelle) [inline CHANGELOG.md:46]:
```suggestion
  For improved compatibility with OpenTelemetry's semantic conventions, the agent's datastore (for databases) and external request (for HTTP clients) segments have been updated with additional attributes.
```

**Review by kaylareopelle — CHANGES_REQUESTED** [review]

**Review by kaylareopelle — APPROVED** [review]

**Pull request merged**